spec/core/hard_machine_spec.rb

```diff
@@ -0,0 +1,83 @@
+require "rspec"
+require "core/game"
+require "core/board"
+require "core/hard_machine"
+
+RSpec.describe HardMachine do
+  it "chooses a winning move when it has the occasion" do
+    move = get_best_move([
+      "X", " ", "X",
+      " ", "O", " ",
+      " ", " ", "O"
+    ])
+
+    expect(move).to eql(1)
+  end
+
+  it "chooses a winning move when it has the occasion (II)" do
+    move = get_best_move([
+      "X", " ", " ",
+      " ", "O", " ",
+      "X", " ", "O"
+    ])
+
+    expect(move).to eql(3)
+  end
+
+  it "chooses a blocking move when the opponent is going to win in the following turn" do
+    move = get_best_move([
+      " ", " ", "X",
+      " ", "O", " ",
+      "X", " ", "O"
+    ])
+
+    expect(move).to eql(0)
+  end
+
+  it "chooses a blocking move when the opponent is going to win in the following turn (II)" do
+    move = get_best_move([
+      " ", " ", "X",
+      "X", " ", "O",
+      "O", " ", "O"
+    ])
+
+    expect(move).to eql(7)
+  end
+
+  it "draws a game against another HardMachine" do
```

Comment:
No value in putting the class name in your test. I'd actually suggest changing the name, but that refactoring is a lot harder since you've encoded your test names with the class name.